doc/fundamentals/history.md

```diff
@@ -0,0 +1,120 @@
+# History
+
+Pachyderm implements rich version-control and history semantics. This doc
+lays out the core concepts and architecture of Pachyderm's version-control
+and the various ways to use the system to access historical data.
+
+## Commits
+
+Commits are the core version-control primitive in Pachyderm, similar to
+git, commits represent an immutable snapshot of a filesystem and can be
+accessed with an ID. Commits have a parentage structure, with new commits
+inheriting content from their parents and then adding to it, you can think
+of it as a linked list, it's also often referred to as "a chain of
+commits." Commit IDs are useful if you want to have a static pointer to
+a snapshot of a filesystem. However, because they're static their use is
+limited, and you'll mostly deal with branches instead.
+
+## Branches
+
+Branches are tags that point to commits, again similar to git, branches
+have semantically meaningful names such as `master` and `staging`.
+Branches are mutable, they move along a growing chain of commits as you
+commit to the branch, and can even be reassigned to any commit within the
+repo (with `create branch`). The commit that a branch points to is
+referred to as the branches "head," and the head's ancestors are referred
+to as "on the branch." Branches can be substituted for commits in
+Pachyderm's API and will behave as if the head of the branch were passed.
+This allows you to deal with semantic meaningful names for commits that
+can be update, rather than static opaque identifiers.
```

Comment:
Will do, btw I normally like to use GH's suggestion feature for fixes like this.

```suggestion
can be updated, rather than static opaque identifiers.
```